### DISCUSSION TITLE
Current error handling mainly involves calling the `onInvalidCandidate` callback function to handle invalid candidates, and printing corresponding error messages to the console.

### DISCUSSION THREAD

**Discussion by lvrocky** [webpack/webpack#18240] [Q&A]
# Bug report

**What is the current behavior?**

1. **Specific Error Messages**:
   - **Problem**: Current error handling mainly involves calling the `onInvalidCandidate` callback function to handle invalid candidates, and printing corresponding error messages to the console.
   - **Optimization Suggestion**: Directly throw an error with specific error messages when encountering invalid candidates, providing better insight into what went wrong.

**What is the expected behavior?**

export function compileCandidates(
  rawCandidates: Iterable<string>,
  designSystem: DesignSystem,
  { onInvalidCandidate }: { onInvalidCandidate?: (candidate: string) => void } = {},
) {
  // ...

  for (let rawCandidate of rawCandidates) {
    let candidate = designSystem.parseCandidate(rawCandidate)
    if (candidate === null) {
      if (onInvalidCandidate) {
        onInvalidCandidate(rawCandidate);
      } else {
        throw new Error(`Invalid candidate: "${rawCandidate}"`);
      }
      continue; // Bail, invalid candidate
    }
    candidates.set(candidate, rawCandidate);
  }

  // ...
}

export function compileAstNodes(rawCandidate: string, designSystem: DesignSystem) {
  let candidate = designSystem.parseCandidate(rawCandidate);
  if (candidate === null) {
    throw new Error(`Invalid candidate: "${rawCandidate}"`);
  }

  // ...
}

**Other relevant information:**
webpack version: 5.90.3
Node.js version: v20.9.0
Operating System: Windows 11 Professional
Additional tools:

**Comment by alexander-akait** [comment]
Sorry, not related to webpack
